services/QuillLMS/app/queries/adapters/csv/csv_data_export.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
```

Comment:
This class shares a lot of similar method implementations and signatures to AdminPremiumDataExport. 

- `row_contains_requested_columns?`, identical
- `BigQueryResultMissingRequestedColumnError`, identical 
- `format_cell`, `to_csv_string`, `validate_input!` similar 

Please consider refactoring this class so that code is less duplicated, via inheritance, module inclusion, or some other method.

Reply:
I've refactored this adapter, and also refactored your upload worker to use the new `UploadToS3` service.